lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+const assert = require('assert');
+const _ = require('lodash');
+const Promise = require('bluebird');
+const GcpStorage = require('@google-cloud/storage');
+const GcpCompute = require('@google-cloud/compute');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class GcpClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(this.settings);
+    this.storageClient = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+    return this.storageClient
+      .bucket(container)
+      .get();
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    const queryOptions = {};
+    if (options && options.prefix) {
+      queryOptions.prefix = options.prefix;
+    }
+    queryOptions.autoPaginate = true;
+    return Promise.try(() =>
+      this.storageClient
+      .bucket(container)
+      .getFiles(queryOptions)
+      .then(results => {
+        const resultFiles = results[0];
```

Comment:
For other IaaS clients, like [AzureClient](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/rel-2017.T12b/lib/iaas/AzureClient.js#L61), we also pick lastModified, which I checked is not getting used. Hence started sending files as an array of objects instead of array of names. I think let's keep this as an array of objects for future needs.

Reply:
The code suggested above by me will still return an array of file names and it does exactly same what is written below from lines 56-63 :-). You can test once if you would like, its just concise and thats how its kept at other places in code base.

Reply:
With little modification it returns as expected array of objects.
```
        return _.get(results, 0, []).map(file => {
          return {name: file.name}
        });
```